### PR TITLE
Add configuration via lsp guide to README

### PR DESCRIPTION
**Pull request by LaurenceWarne**
Hi, this PR is in response to https://github.com/python-lsp/pylsp-mypy/issues/32, I've added a description for how the plugin can be configured (to the best of my understanding :slightly_smiling_face:) using lsp, and given an example similar to that given in https://github.com/python-lsp/python-lsp-ruff#configuration.

Thanks!

### CODE REVIEW

**Comment by Richardk2n** [comment]
This is editor dependent, is it not?
I am not sure if we are responsible to provide documentation for that. In my opinion, the editor should do this.
If we do, we would need to cover a few relevant editors.
The preferred style (sadly badly documented here) is using `pyproject.toml`.
Maybe I should expand on that a little more, but I am not too keen on encouraging the use of esoteric command line editors.

**Comment by LaurenceWarne** [comment]
> I am not sure if we are responsible to provide documentation for that.

Yes it will be editor depedendent, I guess I was mainly trying to address that AFAICS the README config section doesn't mention that configuration from LSP is supported at all.

> Maybe I should expand on that a little more, but I am not too keen on encouraging the use of esoteric command line editors.

I'm happy to drop the example.  How about doing something similar to https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md by sticking the configuration options in a table with the configuration key? e.g.:

| `pyproject.toml` key | LSP configuration key      | Type      | Description                                                                                                                                            | Default |
|----------------------|----------------------------|-----------|--------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
| `live_mode`          | `pylsp.plugins.pylsp_mypy` | `boolean` | This writes to a tempfile every time a check is done. Turning off live_mode means you must save your changes for mypy diagnostics to update correctly. | `True`  |

etc

**Comment by nramirezuy** [comment]
@LaurenceWarne do you know how to handle overrides?

```lua
 pylsp_mypy = {
    enabled = true,
    overrides = {"--python-executable", "python"},
 }
```
This ends up with this:
image

But if I configure it through `pyproject.toml` works as expected. :thinking:

**Comment by Richardk2n** [comment]
@LaurenceWarne This seems sensible to me.

@nramirezuy The `true` is the placeholder for the default command line arguments, that we pass, so depending on what you insert into the `overrides` it is required.
What exactly do you mean by set from `pyprojectl.toml`?

**Comment by LaurenceWarne** [comment]
> This seems sensible to me.

:+1: I've made edits to the PR

File: README.rst [hunk 1]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+     - ``.dmypy.json``
+   * - ``config_sub_paths``
+     - ``pylsp.plugins.pylsp_mypy.config_sub_paths``
+     - ``array`` of ``string`` items
+     - Specifies sub paths under which the mypy configuration file may be found. For each directory searched for the mypy config file, this also searches the sub paths specified here.
+     - ``[]``
+   * - ``report_progress``
+     - ``pylsp.plugins.pylsp_mypy.report_progress``
+     - ``boolean``
+     - Report basic progress to the LSP client. With this option, pylsp-mypy will report when mypy is running, given your editor supports LSP progress reporting. For small files this might produce annoying flashing in your editor, especially in with ``live_mode``. For large projects, enabling this can be helpful to assure yourself whether mypy is still running.
```
Comment (Richardk2n) [inline README.rst:68]: Change 'especially in with `live_mode`' to 'especially in `live_mode`'. This typo was already present, but might as well fix it now.

File: README.rst [hunk 2]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+   * - ``pyproject.toml`` key
+     - LSP Configuration Key
+     - Type
+     - Description
+     - Default
+   * - ``live_mode``
+     - ``pylsp.plugins.pylsp_mypy.live_mode``
+     - ``boolean``
+     - Provides type checking as you type. This writes to a tempfile every time a check is done. Turning off ``live_mode`` means you must save your changes for mypy diagnostics to update correctly.
+     - True
```
Comment (Richardk2n) [inline README.rst:39]: For TOML the `True` actually has to be lower case. It is upper case here due to historic reasons. Given that the LSP config is lower case as well (as far as I know), this is a good chance to change this.

File: README.rst [hunk 3]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+   * - ``dmypy``
+     - ``pylsp.plugins.pylsp_mypy.dmypy``
+     - ``boolean``
+     - Executes via ``dmypy run`` rather than ``mypy``. This uses the ``dmypy`` daemon and may dramatically improve the responsiveness of the ``pylsp`` server, however this currently does not work in ``live_mode``. Enabling this disables ``live_mode``, even for conflicting configs.
+     - False
```
Comment (Richardk2n) [inline README.rst:44]: Boolean → lower case

File: README.rst [hunk 4]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+   * - ``strict``
+     - ``pylsp.plugins.pylsp_mypy.strict``
+     - ``boolean``
+     - Refers to the ``strict`` option of ``mypy``. This option often is too strict to be useful.
+     - False
```
Comment (Richardk2n) [inline README.rst:49]: Boolean → lower case

File: README.rst [hunk 5]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+   * - ``overrides``
+     - ``pylsp.plugins.pylsp_mypy.overrides``
+     - ``array`` of ``string`` items or ``True``
```
Comment (Richardk2n) [inline README.rst:52]: Boolean → lower case
Maybe change to '`array` of (`string` items or `true`)' for clarity.

File: README.rst [hunk 6]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+     - A list of alternate or supplemental command-line options. This modifies the options passed to ``mypy`` or the mypy-specific ones passed to ``dmypy run``. When present, the special boolean member ``True`` is replaced with the command-line options that would've been passed had ``overrides`` not been specified.
+     - ``[True]``
```
Comment (Richardk2n) [inline README.rst:54]: Boolean → lower case

File: README.rst [hunk 7]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+   * - ``config_sub_paths``
+     - ``pylsp.plugins.pylsp_mypy.config_sub_paths``
+     - ``array`` of ``string`` items
+     - Specifies sub paths under which the mypy configuration file may be found. For each directory searched for the mypy config file, this also searches the sub paths specified here.
+     - ``[]``
+   * - ``report_progress``
+     - ``pylsp.plugins.pylsp_mypy.report_progress``
+     - ``boolean``
+     - Report basic progress to the LSP client. With this option, pylsp-mypy will report when mypy is running, given your editor supports LSP progress reporting. For small files this might produce annoying flashing in your editor, especially in with ``live_mode``. For large projects, enabling this can be helpful to assure yourself whether mypy is still running.
+     - False
```
Comment (Richardk2n) [inline README.rst:69]: Boolean → lower case

**Review by Richardk2n — CHANGES_REQUESTED** [review]
Maybe make the short summary in every description bold.

**Comment by LaurenceWarne** [comment]
I think I've addressed everything :+1:

File: README.rst [hunk 8]
```diff
@@ -24,29 +24,54 @@ Install into the same virtualenv as python-lsp-server itself.
...
+     - LSP Configuration Key
+     - Type
+     - Description
+     - Default
+   * - ``live_mode``
+     - ``pylsp.plugins.pylsp_mypy.live_mode``
+     - ``boolean``
+     - **Provides type checking as you type**. This writes to a tempfile every time a check is done. Turning off ``live_mode`` means you must save your changes for mypy diagnostics to update correctly.
+     - true
+   ** - ``dmypy``
```
Comment (Richardk2n) [inline README.rst:40]: extra *

**Review by Richardk2n — APPROVED** [review]

**Comment by nramirezuy** [comment]
> @nramirezuy The `true` is the placeholder for the default command line arguments, that we pass, so depending on what you insert into the `overrides` it is required. What exactly do you mean by set from `pyprojectl.toml`?

Sorry I never came back to you, I meant setting the executable from `pyproject.toml` with [this option](https://mypy.readthedocs.io/en/stable/config_file.html#confval-python_executable).

So my configuration ended up like [this](https://github.com/nramirezuy/dotfiles/blob/main/src/tag-nvim/config/nvim/lua/plugins/lspconfig/python.lua#L50), which I think should be the default, since it works with [pyenv](https://github.com/pyenv/pyenv) and [venv](https://docs.python.org/3/library/venv.html).

**I'm not looking to start a discussion on a closed ticket**, but I'm happy to be @ in a new ticket about the subject.

**Pull request merged**